feat: add liatrio distro of otel run optionally for delivery metrics (#1)
liatrio/opentelemetry-demo: docs/delivery.md

# Delivery Demo

> IMPORTANT: At this time, this fork only supports the Docker Setup.

This bonus section showcases delivery focused metrics by leveraging the Astronomy
shop. There are two sets of delivery metrics available through this fork:

1. Leading indicators (GitHub Metrics)
2. Lagging indicators (DORA from GitHub)

Both sets of metrics are meant to be leveraged together, yet on their own, still
provide engineers with valuable data. These metrics capture point-in-time data and

> are meant to be looked at overtime to determine trends of improvemenent.

Reviewer: ```suggestion
are meant to be looked at overtime to determine trends of improvement.
```

There is some setup required in order to get these metrics working. You will need:

> - A personal access token with access to GitHub 

Reviewer: ```suggestion
- A personal access token with access to GitHub
```

> - A repository setup to emit even logs from GitHub through a webhook based GitHub

Reviewer: ```suggestion
- A repository setup to emit event logs from GitHub through a webhook based GitHub
```

app
- Some mechanism to forward webhook calls to your local machine, like ngrok

We'll detail how to setup these things up exactly in a later section.

> **Table of Contents**

Reviewer: ```suggestion
## Table of Contents

```

- [Delivery Demo](#delivery-demo)
    - [Architecture](#architecture)
    - [Pre-requisites](#pre-requisites)
        - [PAT for GitHub Metrics](#personal-access-token-for-github-metrics)
        - [GitHub App Setup](#github-app-setup-for-webhook-events)
    - [Local Docker Setup](#local-docker-setup)
        - [Leading Indicators](#leading-delivery-indicators-github)
        - [Lagging Indicators (DORA)](#lagging-delivery-indicators-dora)

## Architecture

> To delivery metrics side of the house relies on the existing architecture of the
> demo but adds in the [Liatrio distribution of the OpenTelemetry](https://github.com/liatrio/liatrio-otel-collector)
> collector to gather metrics from GitHub through the [Git Provider Receiver](https://github.com/liatrio/liatrio-otel-collector/tree/main/receiver/gitproviderreceiver)
> and the [WebHook Event Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/webhookeventreceiver)
> which gathers event logs to populate DORA metrics.

Reviewer: ```suggestion
The `delivery metrics` demo expands on the existing OpenTelemetry demo by using the
[Liatrio distribution of OpenTelemetry](https://github.com/liatrio/liatrio-otel-collector)
collector to gather metrics from GitHub through the [Git Provider
Receiver](https://github.com/liatrio/liatrio-otel-collector/tree/main/receiver/gitproviderreceiver)
and the [WebHook Event Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/webhookeventreceiver).
This gathered event logs are used populate a DORA metrics dashboard in Grafana.
```
